cmd/pilot-discovery/main.go

```diff
@@ -194,11 +227,17 @@ func init() {
 		"Enable profiling via web interface host:port/debug/pprof")
 	discoveryCmd.PersistentFlags().BoolVar(&flags.discoveryOptions.EnableCaching, "discovery_cache", true,
 		"Enable caching discovery service responses")
-	discoveryCmd.PersistentFlags().StringVar(&flags.consulargs.config, "consulconfig", "",
-		"Consul Config file for discovery")
-	discoveryCmd.PersistentFlags().StringVar(&flags.consulargs.serverURL, "consulserverURL", "",
-		"URL for the consul server")
 
+	discoveryCmd.PersistentFlags().StringVar(&flags.consul.config, "consulconfig", "",
+		"Consul Config file for discovery")
+	discoveryCmd.PersistentFlags().StringVar(&flags.consul.serverURL, "consulserverURL", "",
+		"URL for the Consul server")
+	discoveryCmd.PersistentFlags().StringVar(&flags.eureka.serverURL, "eurekaserverURL", "",
+		"URL for the Eureka server")
+
+	_ = discoveryCmd.PersistentFlags().MarkHidden("namespace")
```

Comment:
let's not hide flags. You can rename it to a longer name if you worry they might confuse on consul.

Reply:
These are dev flags.. not for public consumption on k8s or consul..